Sources/Build/BuildDelegate.swift

```diff
@@ -0,0 +1,223 @@
+    /// to identify the process. This handle should only be used to associate
+    /// different status calls relating to the same process. It is only
+    /// guaranteed to be unique from when it has been provided here to when it
+    /// has been provided to the \see commandProcessFinished() call.
+    public func commandProcessStarted(_ command: llbuildSwift.Command, process: ProcessHandle) {
+    }
+
+    /// Called to report an error in the management of a command process.
+    ///
+    /// - parameter process: The process handle.
+    /// - parameter message: The error message.
+    public func commandProcessHadError(_ command: llbuildSwift.Command, process: ProcessHandle, message: String) {
+    }
+
+    /// Called to report a command processes' (merged) standard output and error.
+    ///
+    /// - parameter process: The process handle.
+    /// - parameter data: The process output.
+    public func commandProcessHadOutput(_ command: llbuildSwift.Command, process: ProcessHandle, data: [UInt8]) {
+        outputStream <<< data
+    }
+
+    /// Called when a command's job has finished executing an external process.
+    ///
+    /// - parameter process: The handle used to identify the process. This
+    /// handle will become invalid as soon as the client returns from this API
+    /// call.
+    ///
+    /// - parameter result: Whether the process suceeded, failed or was cancelled.
+    /// - parameter exitStatus: The raw exit status of the process.
+    public func commandProcessFinished(
+        _ command: llbuildSwift.Command,
+        process: ProcessHandle,
+        result: CommandExtendedResult
+    ) {
+    }
+
+    /// Called when a cycle is detected by the build engine and it cannot make
+    /// forward progress.
+    public func cycleDetected(rules: [BuildKey]) {
+        diagnostics.emit(data: LLBuildCycleErrorDiagnostic(rules: rules))
+    }
+
+    /// Called when a cycle is detected by the build engine to check if it should
+    /// attempt to resolve the cycle and continue
+    ///
+    /// - parameter rules: The ordered list of items comprising the cycle,
+    /// starting from the node which was requested to build and ending with the
+    /// first node in the cycle (i.e., the node participating in the cycle will
+    /// appear twice).
+    /// - parameter candidate: The rule the engine will use to attempt to break the
+    /// cycle.
+    /// - parameter action: The action the engine will take on the candidateRule.
+    ///
+    /// Returns true if the engine should attempt to resolve the cycle, false
+    /// otherwise. Resolution is attempted by either forcing items to be built, or
+    /// supplying a previously built result to a node in the cycle. The latter
+    /// action may yield unexpected results and thus this should be opted into
+    /// with care.
+    public func shouldResolveCycle(rules: [BuildKey], candidate: BuildKey, action: CycleAction) -> Bool {
+        return true
```

Comment:
This should *definitely* return false. Asking llbuild to break cycles is a risky thing we don't usually want to do.

Reply:
Done